docs/src/tutorial.jl

```diff
@@ -708,161 +707,87 @@ end
 # a function that calls other functions depending on the type, such as
 
-function union_step!(agent, model)
-    if typeof(agent) <: AgentSchelling
-        schelling_step!(agent, model)
-    elseif typeof(agent) <: Politician
-        politician_step!(agent, model)
-    end
-end
-
-# and then passing
-
-model = StandardABM(
-    Union{SchellingAgent, Politician}, # type of agents
-    space; # space they live in
-    agent_step! = union_step!
-)
-
-# This approach also works with the [`@multiagent`](@ref) possibility we discuss below.
-# `Union` types however also offer the unique possibility of utilizing fully the Julia's
-# [multiple dispatch system](https://docs.julialang.org/en/v1/manual/methods/).
-# Hence, we can use the same function name and add dispatch to it, such as:
-
-function dispatch_step!(agent::SchellingAgent, model)
+function agent_step!(agent::Schelling, model)
     ## stuff.
 end
 
-function dispatch_step!(agent::Politician, model)
+function agent_step!(agent::Politician, model)
     ## other stuff.
 end
 
-# and give `dispatch_step!` to the `agent_step!` keyword during model creation.
-
-# ## Multiple agent types with `@multiagent`
-
-# [`@multiagent`](@ref) is a macro, and hence not "native Julia" syntax like `Union`,
-# however it has been designed to be as similar to [`@agent`](@ref) as possible.
-# The syntax to use it is like so:
-
-@multiagent struct MultiSchelling{X}(GridAgent{2})
-    @subagent struct Civilian # can't re-define existing `Schelling` name
-        mood::Bool = false
-        group::Int
-    end
-    @subagent struct Governor{X} # can't redefine existing `Politician` name
-        group::Int
-        influence::X
-    end
-end
-
-# This macro created three names into scope:
-
-(MultiSchelling, Civilian, Governor)
-
-# however, only one of these names is an actual Julia type:
-
-fieldnames(MultiSchelling)
-
-# that contains all fields of all subtypes without duplication, while
+# and then passing
 
-fieldnames(Civilian)
+model = StandardABM(
+    Union{Schelling, Politician}, # type of agents
+    space; # space they live in
+    agent_step!
+)
 
-# doesn't have any fields. Instead,
-# you should think of `Civilian` and `Governor` as just convenience functions that have been
-# defined for you to "behave like" types. That's why we call these **kinds** and not
-# **types**.
+# ## Multiple agent types with `@sumtype`
 
-# E.g., you can initialize
+# By using `@sumtype` from `DynamicSumTypes.jl` it is possible to improve the 
+# computational performance of simulations requiring multiple types, while almost
+# everything works the same 
 
-civ = Civilian(; id = 2, pos = (2, 2), group = 2) # default `mood`
+using DynamicSumTypes
 
-# or
+@sumtype MultiSchelling(Schelling, Politician) <: AbstractAgent
 
-gov = Governor(; id = 3 , pos = (2, 2), group = 2, influence = 0.5)
+# Now when you create instances you will need to enclose them in `MultiSchelling`
```

Comment:
This is a clear and definitive disadvantage over `@multiagent`. That you have to always create a type of this Multi Agent type, instead of the standard Agent type. It really adds one more step versus the single agent usage we had so far in both 1-type models, Union type, or `@multiagent` type.

Have you thought how this impacts things on the user side?